gcc/d/dfrontend/expression.c

```diff
@@ -4082,6 +4082,12 @@ Expression *ArrayLiteralExp::semantic(Scope *sc)
         return new ErrorExp();
     }
 
+    if (global.params.noTypeinfo && elements->dim != 0)
+    {
+        error (global.params.noTypeinfo, "Can't use array literal");
```

Comment:
Static arrays should be ok here without RTTI.  GDC and LDC at least are smart enough to not ``new`` an array literal that will ultimately land in a static array.  I hope DMD is not the odd one out - @yebblies  ?